specification.md

```diff
@@ -0,0 +1,344 @@
+        -   (c) May require the user to specify type multiple times with
+            different field names, e.g. with "@context" for
+            [JSON-LD](https://json-ld.org/).
+    2.  It would incur double base64 encoding overhead for non-JSON payloads.
+    3.  It is more complex than PAE.
+
+# Backwards Compatibility
+
+## Current format
+
+The
+[current signature format](https://github.com/in-toto/docs/blob/master/in-toto-spec.md#42-file-formats-general-principles)
+used by TUF and in-toto has a BODY that is a regular JSON object and a signature over the
+[Canonical JSON] serialization of BODY.
+
+```json
+{
+  "signed": <BODY>,
+  "signatures": [{
+    …,
+    "sig": "<Hex(Sign(CanonicalJson(BODY)))>"
+  }, …]
+}
+```
+
+To verify, the consumer parses the whole JSON file, re-serializes BODY using
+Canonical JSON, then verifies the signature.
+
+## Detect if a document is using old format
+
+To detect whether a signature is in the old or new format:
+
+-   If it contains a `payload` field, assume it is in the new format.
+-   If it contains a `signed` field, assume it is in the old format.
+
+To convert an existing signature to the new format:
+
+-   `new.payload = base64encode(CanonicalJson(orig.signed))`
+-   `new.payloadType = "<URI>/backwards-compatible-json"`
+-   `new.signatures[*].sig = base64encode(hexdecode(orig.signatures[*].sig))`
+
+To convert a backwards compatible signature to the old format:
+
+-   `old.signed = jsonparse(base64decode(new.payload))`
+-   `old.signatures[*].sig = hexencode(base64decode(new.signatures[*].sig))`
+
+# Security
+
+# Testing
+
+See [reference implementation](reference_implementation.ipynb). Here is an
+example.
+
+BODY:
+
+```none
+hello world
+```
+
+PAYLOAD_TYPE:
+
+```none
+http://example.com/HelloWorld
+```
+
+PAE:
+
+```none
+02 00 00 00 00 00 00 00 1d 00 00 00 00 00 00 00
+68 74 74 70 3a 2f 2f 65 78 61 6d 70 6c 65 2e 63
+6f 6d 2f 48 65 6c 6c 6f 57 6f 72 6c 64 0b 00 00
+00 00 00 00 00 68 65 6c 6c 6f 20 77 6f 72 6c 64
+```
+
+Cryptographic keys:
+
+```none
+Algorithm: ECDSA over NIST P-256 and SHA-256, with deterministic-rfc6979
+Signature: raw concatenation of r and s (Cryptodome binary encoding)
+X: 46950820868899156662930047687818585632848591499744589407958293238635476079160
+Y: 5640078356564379163099075877009565129882514886557779369047442380624545832820
+d: 97358161215184420915383655311931858321456579547487070936769975997791359926199
+```
+
+Signed wrapper:
+
+```json
+{"payload": "aGVsbG8gd29ybGQ=",
+ "payloadType": "http://example.com/HelloWorld",
+ "signatures": [{"sig": "y7BK8Mm8Mr4gxk4+G9X3BD1iBc/vVVuJuV4ubmsEK4m/8MhQOOS26ejx+weIjyAx8VjYoZRPpoXSNjHEzdE7nQ=="}]}
+```
+
+# Infrastructure Requirements
+
+# References
+
+[Canonical JSON]: http://gibson042.github.io/canonicaljson-spec/
+[JWS]: https://tools.ietf.org/html/rfc7515
+[PASETO]: https://github.com/paragonie/paseto/blob/master/docs/01-Protocol-Versions/Version2.md#sig
```

Comment:
FYI, these links are [not rendered here](https://github.com/secure-systems-lab/signing-spec/blob/43b1aa834ff0242c20988d339206c598bac25170/specification.md#references)